openedx/core/djangoapps/user_api/accounts/tests/pact/PactBypassAuthMiddleware.py

```diff
@@ -0,0 +1,36 @@
+"""
+Contain the middleware logic needed during pact verification
+"""
+from django.contrib import auth
+from django.utils.deprecation import MiddlewareMixin
+from django.conf import settings
+from django.core.exceptions import MiddlewareNotUsed
+
+User = auth.get_user_model()
+
+
+class AuthenticationMiddleware(MiddlewareMixin):
+    """
+    Middleware to add default authentication into the requests for pact verification.
+
+    This middleware is required to add a default authenticated user and bypass CSRF validation
+    into the requests during the pact verification workflow. Without the authentication, the pact verification
+    process will not work as the apis.
+    See https://docs.pact.io/faq#how-do-i-test-oauth-or-other-security-headers
+    """
+
+    def __init__(self, get_response):
+        if getattr(settings, 'PACT_BYPASS_AUTHENTICATION', None) is False:
```

Comment:
If the setting is missing, you'll get `None`; `None is False` evaluates to false, so the middleware won't be skipped.

If you instead use `not`, then None and False will both be treated as false-ish conditions:

```suggestion
        if not getattr(settings, 'PACT_BYPASS_AUTHENTICATION', None):
```

Reply:
Ah, I see. Just made the switch.